Storage/SQL/SQLiteBookmarksBase.swift

```diff
@@ -35,4 +45,21 @@ public class SQLiteBookmarks: BookmarksModelFactorySource {
         
         return self.db.queryReturnsResults(sql, args: args)
     }
+
+    public func getURLForKeywordSearch(keyword: String) -> Deferred<Maybe<String>> {
+        let sql = "SELECT bmkUri FROM \(TableBookmarksBuffer) WHERE " +
```

Comment:
@sleroux, can you help me with the database migration? I think we need to add an ALTER TABLE command to add the new columns to the migration script.